chain/walker.go

```diff
@@ -90,6 +90,10 @@ func (c *Walker) WalkChain(ctx context.Context, node lens.API, ts *types.TipSet)
 			return xerrors.Errorf("get tipset: %w", err)
 		}
 
+		if int64(ts.Height()) < c.minHeight {
```

Comment:
This prevents tasks that require a diff between parent and child to stop processing at the lower bound of the walk. I ran the same walk (`visor walk --from 1007052 --to 1007054`) before and after this change.

Before this change I got processing reports for all tasks for heights 1007052, 1007053 and 1007054, plus 3 for 1007051 (blocks, chaineconomics, implicitmessages)

After the change I only got reports from all tasks for heights 1007053 and 1007054, and 3 for 1007052.

I'm going to suggest a different change to avoiding overshoot in a new PR. Let's keep this one focused on off-by-one in the processing report table. But note that https://github.com/filecoin-project/sentinel-visor/pull/650 also affects this.

Reply:
Also this change affects the handling of null rounds by the consensus task. If the low bound height happens to be a null round then the consensus task will never see it since the closest ancestor tipset will have a height lower than the minimum.

This actually needs some thought in general. We may need to pass bounds into the indexer so that it can allow the consensus task to report null rounds that extend back or forward from a tipset to the bounds of the walk.